app/controllers/vouchers_controller.rb

```diff
@@ -58,17 +58,49 @@ def create
       @existing_voucher = @swap.vouchers.find_by(client_id: @voucher.client_id)
       return render :new
     end
-
+    
     if !@voucher.save
       return render :new
     end
 
+
     redirect_to action: :created, id: @voucher.id
   end
 
   def created
   end
 
+  def send_voucher
+    # EMAIL
+    if params[:commit] == 'E-mail'
+      @voucher = Voucher.find(params[:id])  
```

Comment:
this happens in both the email and sms branches, can we move those lines up and collapse into one call?